process/block/preprocess/scheduledTxsExecution.go

```diff
@@ -216,29 +226,53 @@ func (ste *scheduledTxsExecution) GetScheduledSCRs() map[block.Type][]data.Trans
 	return mapScheduledSCRs
 }
 
-// SetScheduledSCRs sets the given scheduled SCRs
-func (ste *scheduledTxsExecution) SetScheduledSCRs(mapScheduledSCRs map[block.Type][]data.TransactionHandler) {
+// SetScheduledSCRs sets the resulted SCRs after the execution of scheduled transactions
+func (ste *scheduledTxsExecution) SetScheduledSCRs(mapSCRs map[block.Type][]data.TransactionHandler) {
 	ste.mutScheduledTxs.Lock()
 	defer ste.mutScheduledTxs.Unlock()
 
 	numScheduledSCRs := 0
 	ste.mapScheduledSCRs = make(map[block.Type][]data.TransactionHandler)
-	for blockType, scheduledSCRs := range mapScheduledSCRs {
-		if len(scheduledSCRs) == 0 {
+	for blockType, scrs := range mapSCRs {
+		if len(scrs) == 0 {
 			continue
 		}
 
-		ste.mapScheduledSCRs[blockType] = make([]data.TransactionHandler, len(scheduledSCRs))
-		for scrIndex, txHandler := range scheduledSCRs {
+		ste.mapScheduledSCRs[blockType] = make([]data.TransactionHandler, len(scrs))
+		for scrIndex, txHandler := range scrs {
 			ste.mapScheduledSCRs[blockType][scrIndex] = txHandler
 		}
 
-		numScheduledSCRs += len(scheduledSCRs)
+		for scrIndex := range scrs {
```

Comment:
You have this print a few times in this file. Maybe do a separate function for it. And inside verify if log.trace is set before doing the range.

Reply:
Changed